components/chassis.py

```diff
@@ -108,23 +119,32 @@ def __init__(
         drive_motor_config.neutral_mode = NeutralModeValue.BRAKE
         drive_motor_config.inverted = (
             InvertedValue.CLOCKWISE_POSITIVE
-            if drive_reversed
+            if config.reverse_drive
             else InvertedValue.COUNTER_CLOCKWISE_POSITIVE
         )
 
         drive_gear_ratio_config = FeedbackConfigs().with_sensor_to_mechanism_ratio(
-            1 / self.DRIVE_MOTOR_REV_TO_METRES
+            1 / (config.wheel_circumference * config.drive_ratio)
         )
 
         # configuration for motor pid and feedforward
-        self.drive_pid = Slot0Configs().with_k_p(1.0868).with_k_i(0).with_k_d(0)
-        self.drive_ff = SimpleMotorFeedforwardMeters(kS=0.15172, kV=2.8305, kA=0.082659)
+        self.drive_pid = (
+            Slot0Configs()
+            .with_k_p(config.drive_gains.k_p)
+            .with_k_i(config.drive_gains.k_i)
+            .with_k_d(config.drive_gains.k_d)
+        )
+        self.drive_ff = SimpleMotorFeedforwardMeters(
+            kS=config.drive_gains.k_s,
+            kV=config.drive_gains.k_v,
+            kA=config.drive_gains.k_a,
+        )
 
         drive_config.apply(drive_motor_config)
         drive_config.apply(self.drive_pid, 0.01)
         drive_config.apply(drive_gear_ratio_config)
 
-        self.central_angle = Rotation2d(x, y)
+        self.central_angle = Rotation2d(position.x, position.y)
```

Comment:
Isn't there a Translation2d method to do this these days?
```suggestion
        self.central_angle = position.angle()
```